src/gsi/get_gefs_for_regional.f90

```diff
@@ -336,16 +354,117 @@ subroutine get_gefs_for_regional
      if(mype == 0) then
         write(6,*) ' nemsio: fhour,idate=',fhour,idate
         write(6,*) ' iadate(y,m,d,hr,min)=',iadate
-        write(6,*) ' nemsio: jcap,levs=',njcap,levs
+        write(6,*) ' nemsio: jcap,levs=',levs
         write(6,*) ' nemsio: latb,lonb=',latb,lonb
         write(6,*) ' nemsio: idvc,nvcoord=',idvc,nvcoord
         write(6,*) ' nemsio: idsl=',idsl
      end if
 
+! LB: add netCDF header information: 
+     else ! use_gfs_ncio and get this information
+        write(6,*) 'LB netcdf test0'
+        write(sfilename,'("sfcf",i2.2)')nhr_assimilation
+        write(6,*) ' LB netcdf test1'
+        write(6,*) ' LB netcdf nhr',nhr_assimilation
+        ! open the netCDF file
+        atmges = open_dataset(filename)
+        !sfcges = open_dataset(sfilename)
+        write(6,*) ' LB netcdf test2'
+        ! get dimension sizes
+        ncdim = get_dim(atmges, 'grid_xt'); gfshead%lonb = ncdim%len
+        ncdim = get_dim(atmges, 'grid_yt'); gfshead%latb = ncdim%len
+        ncdim = get_dim(atmges, 'pfull') ; gfshead%levs = ncdim%len
+        write(6,*) 'region lonb latb=',gfshead%lonb, gfshead%latb,gfshead%levs         
+        ! LB: add nsig_gfs here: 
+        nsig_gfs = gfshead%levs
+        ! hard code jcap,idsl,idvc
+        gfshead%jcap = -9999
+        gfshead%idsl= 1
+        gfshead%idvc = 2
+
+        ! FV3GFS write component does not include JCAP, infer from DIMY-2
+        if (njcap<0) njcap=latb-2
+
+        nlat_gfs=gfshead%latb+2
+        nlon_gfs=gfshead%lonb
+        nsig_gfs=gfshead%levs
+        write(6,*) 'jcap', jcap_ens
+        write(6,*) 'jcap', gfshead%jcap
+        write(6,*) 'jcap1'
+
+!         jcap_gfs=latb-2
+         jcap_gfs=gfshead%latb-2
+
+!        if(gfshead%jcap > 0)then
+!          jcap_gfs=gfshead%jcap
+!        else if(jcap_ens > 0)then
+!          jcap_gfs=jcap_ens
+!        else
+!          write(6,*)'get_gefs_for_regional:ERROR ncio jcap is undefined'
+!        endif
+
+        write(6,*) 'regional jcap_gfs=',jcap_gfs
+
+
+!        call read_attribute(atmges, 'ncnsto', ntrac)
```

Comment:
If these lines aren't needed, remove the commented lines.

Reply:
Removed.